text/0000-rust-has-provenance.md

```diff
@@ -0,0 +1,316 @@
+- Feature Name: rust_has_provenance
+- Start Date: 2023-11-22
+- RFC PR: [rust-lang/rfcs#0000](https://github.com/rust-lang/rfcs/pull/0000)
+- Rust Issue: [rust-lang/rust#0000](https://github.com/rust-lang/rust/issues/0000)
+
+# Summary
+[summary]: #summary
+
+Pointers (this includes values of reference type) in Rust have **two** components.
+* The pointer's "address" says where in memory the pointer is currently pointing.
+* The pointer's "provenance" says where in memory the pointer is allowed to access when.
```

Comment:
```suggestion
* The pointer's "provenance" says where in memory the pointer is allowed to access, and when.
```
this was flagged as confusing wording on [HN](https://news.ycombinator.com/item?id=39213815)

Reply:
That comma is not required.

However, you might find this wording to be clearer:
```suggestion
* The pointer's "provenance" says where and when the pointer is allowed to access memory.
```

Reply:
The `and` is also not required, the original sentence was grammatically correct. But the comma and `and` adds emphasis and clarifies the structure of the sentence.